androidApp/src/main/java/dev/johnoreilly/confetti/account/AccountIcon.kt

```diff
@@ -84,23 +114,43 @@ fun AccountIcon(
                 onSwitchConference()
             }
         )
-        if (uiState.showInstallOnWear) {
+        if (showInstallOnWear) {
```

Comment:
Maybe a personal preference but I'd pass the uiState in and create it in the previews.

Reply:
As I wrote in the description, I was not 100% sure, but what is the reason that you would have chosen that approach? :)

Reply:
I'm almost sure I saw a Google video advocating for individual parameters but can't find it anymore. Either works for me

Reply:
I don't think it's the case.  I'll try to find clear guidance.

A few reasons

1) It's really unwieldy, and doesn't scale well as features get more complex.  You end up changing a lot of params up and down the call chain. The Composable contract is also specific to the current UI, so it must change and break callers every time you add some additional field.
2) UiState might be simple now, but over time, you might end up with a union of types.  Loading, Failed, SignedIn, Anonymous.  If these each have different valid params, then your composable with individual fields, every param becomes nullable.
3) to me it reads cleaner

Reply:
I feel like they are generally pretty consistent about this

https://github.com/android/nowinandroid/blob/2388d4eb2417695c41fabd3fa916d899614ccd9f/feature/interests/src/main/java/com/google/samples/apps/nowinandroid/feature/interests/InterestsScreen.kt#L55

XScreen, XUiState, XRoute.

Although they do pick out bits of that UIState and pass it into the individual screen components.  But that makes sense, you only need part of data.

Reply:
> XScreen, XUiState, XRoute

+1 on this naming, makes a lot of sense to me. 

In that very specific `AccountIcon` case it doesn't really apply because `AccountIcon` is not a screen. Having a ViewModel there makes it impossible to preview `SessionsView` for an example (which should really have been `SessionsScreen`).

Question: should we juste remove the `AccountViewModel` altogether and move that state to the upmost "Route" level?

Reply:
I think so.

Reply:
I can fix this, do you want me to do that in this PR? Or is it better to separate the removal of the ViewModel? :)